### PR TITLE
fix(manager/sbt): Support for updating Scala 3

### PR DESCRIPTION
**Pull request by aktowns**
## Changes

Checks if the current Scala version is 3 and if so uses the correct corresponding `scala-library` for determining the latest version.

This change only fixes 3.x updates if you're already on 3.x (so 2 -> 3 isn't offered).

## Context

The underlying `scala-library` name has changed for Scala 3, this breaks how updating the Scala version is currently handled in renovate. This causes version look ups for people on 3.x to be ignored. 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/sbt/extract.ts [hunk 1]
```diff
@@ -49,10 +51,17 @@ const scalaVersionMatch = q
   )
   .handler((ctx) => {
     if (ctx.scalaVersion) {
+      const version = get(mavenVersioning.id);
+
+      let packageName = 'org.scala-lang:scala-library';
+      if (version.getMajor(ctx.scalaVersion) === 3) {
```
Comment (rarkins) [inline lib/modules/manager/sbt/extract.ts:57]: Need to check it's a valid version first

Comment (zharinov) [reply lib/modules/manager/sbt/extract.ts:57]: If it doesn't throw, we're safe without validity check I think

**Review by zharinov — DISMISSED** [review]

File: lib/modules/manager/sbt/extract.ts (hunk 1 above)
Comment (aktowns) [reply lib/modules/manager/sbt/extract.ts:57]: As far as I can tell its not throwing, get back `null` in the event it fails an `isVersion` check https://github.com/renovatebot/renovate/blob/main/lib/modules/versioning/maven/index.ts#LL75C5-L75C5 (which in itself is doing a bunch of regex checks https://github.com/renovatebot/renovate/blob/main/lib/modules/versioning/maven/compare.ts#L280)

**Review by rarkins — DISMISSED** [review]

File: lib/modules/manager/sbt/extract.spec.ts [hunk 2]
```diff
@@ -223,6 +223,21 @@ describe('modules/manager/sbt/extract', () => {
       });
     });
 
+    it('extracts correct scala library when dealing with scala 3', () => {
+      const content = `
+        scalaVersion := "3.1.1"
+      `;
+
+      expect(extractPackageFile(content)).toMatchSnapshot({
```
Comment (viceice) [inline lib/modules/manager/sbt/extract.spec.ts:231]:
```suggestion
      expect(extractPackageFile(content)). toMatchObject({
```

**Review by viceice — CHANGES_REQUESTED** [review]
v2-v3 upgrade can be handled by our replace feature

**Comment by rarkins** [comment]
> v2-v3 upgrade can be handled by our replace feature

I didn't think so because `packageName` is provided by _our extract_ and is not embedded in the package file.

**Review by rarkins — DISMISSED** [review]

File: lib/modules/manager/sbt/__snapshots__/extract.spec.ts.snap [hunk 3]
```diff
@@ -104,6 +104,24 @@ exports[`modules/manager/sbt/extract extractPackageFile() extract deps from nati
...
+      "registryUrls": [
+        "https://repo.maven.apache.org/maven2",
+      ],
+      "separateMinorPatch": true,
+    },
+  ],
+  "packageFileVersion": undefined,
+}
+`;
+
```
Comment (viceice) [inline lib/modules/manager/sbt/__snapshots__/extract.spec.ts.snap:124]:
```suggestion
```

**Review by viceice — DISMISSED** [review]

**Review by rarkins — APPROVED** [review]

File: lib/modules/manager/sbt/extract.spec.ts [hunk 4]
```diff
@@ -223,6 +223,21 @@ describe('modules/manager/sbt/extract', () => {
       });
     });
 
+    it('extracts correct scala library when dealing with scala 3', () => {
+      const content = `
```
Comment (viceice) [inline lib/modules/manager/sbt/extract.spec.ts:227]:
```suggestion
      const content = codeBlock`
```

needs prettier fix

**Review by viceice — CHANGES_REQUESTED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**